lib/services/user.ts

```diff
@@ -8,3 +8,28 @@ export async function findOrCreateUser(email: string): Promise<User> {
     update: {},
   });
 }
+
+export type UserWithPosts = Prisma.PromiseReturnType<typeof getUserWithPosts>;
+
+export async function getUserWithPosts(id: string) {
+  return prisma.user.findUnique({
+    where: { id },
+    include: { posts: true },
+  });
+}
+
+export type ModifiableUser = Pick<User, "name" | "imageUrl">;
```

Comment:
I'm new to this notation. Is there any advantage over writing this out in full as its own type definition?

Reply:
It ensures that the type is a subset of `User`, which is desired in this case

Reply:
Okay. Thanks!